src/pages/workspaces/workspace/analysis/modals/ComputeModal/GCPPersistentDiskInput.ts

```diff
@@ -0,0 +1,38 @@
+import { div } from 'react-hyperscript-helpers';
+import { GCPPersistentDiskOptions, SharedPdType } from 'src/libs/ajax/leonardo/models/disk-models';
+import { GCPPersistentDiskSizeNumberInput } from 'src/pages/workspaces/workspace/analysis/modals/ComputeModal/GCPPersistentDiskSizeNumberInput';
+import { PersistentDiskTypeInputContainer } from 'src/pages/workspaces/workspace/analysis/modals/ComputeModal/PersistentDiskTypeInputContainer';
+
+export interface GCPPersistentDiskInputProps {
+  persistentDiskType: SharedPdType;
+  persistentDiskSize: number;
+  onChangePersistentDiskType: (e: SharedPdType) => void;
+  onChangePersistentDiskSize: (e: number) => void;
+  persistentDiskExists: boolean;
+}
+
+export const GCPPersistentDiskInput = (props: GCPPersistentDiskInputProps) => {
```

Comment:
adding React.FC<Props> type here and for other FCs you are defining will make things more robust, like adding knowledge of key prop.

`export const GCPPresistentDiskInput: React.FC<GCPPersistentDiskInputProps> = ....`